### PR TITLE
Add virchow patch token extraction for dense prediction tasks

### PR DESCRIPTION
**Pull request by michellestegem**

### CODE REVIEW

File: src/unicorn_baseline/vision/pathology/models.py [hunk 1]
```diff
@@ -81,3 +88,63 @@ def forward_slide(self, tile_features, tile_coordinates, tile_size_lv0):
...
+    """
+
+    def __init__(self, model_dir, input_size=224):
+        super().__init__()
+        self.model_dir = model_dir
+        self.input_size = input_size
+        self.device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
+
+        # Load model configuration
+        with open(os.path.join(self.model_dir, "virchow-config.json"), "r") as f:
```
Comment (clemsgrs) [inline src/unicorn_baseline/vision/pathology/models.py:105]: do we want to load `virchow-config.json` from model dir or from `src/unicorn_baseline/vision_language/virchow/virchow_config.json` given we have the file there and it's not used

i would be in for keep loading from model_dir but then removing the file above

File: src/unicorn_baseline/vision/pathology/models.py [hunk 2]
```diff
@@ -81,3 +88,63 @@ def forward_slide(self, tile_features, tile_coordinates, tile_size_lv0):
...
+        self.tile_encoder.eval()
+
+    def get_transforms(self):
+        """Retrieve the transformation pipeline for input images."""
+        data_config = resolve_data_config(
+            self.config["pretrained_cfg"], model=self.tile_encoder
+        )
+        return create_transform(**data_config)
+
+    def forward(self, x):
```
Comment (clemsgrs) [inline src/unicorn_baseline/vision/pathology/models.py:141]: i would have a single Virchow class that handles all three possible forward scenario:
- if self.return_patch_tokens, simply return patch_tokens
- elif self.mode == "full", return torch.cat([class_token, patch_tokens.mean(1)], dim=-1)
- else, return class_token

**Review by clemsgrs — COMMENTED** [review]
left a suggestion to make the Virchow class more generic as we're going to use Virchow with mode="full" for PRISM (image-level vision tasks + vision-language)
